dotnet/aspnet/webappWithTests/Application/SampleWebApplication.FunctionalTests/SampleFunctionalTests.cs

```diff
@@ -18,23 +19,43 @@ public static void Initialize(TestContext testContext)
             SampleFunctionalTests.testContext = testContext;
         }
 
+        [TestInitialize]
+        public void TestInit()
+        {
+            driver = GetChromeDriver();
+        }
+
+        [TestCleanup]
+        public void TestClean()
+        {
+            driver.Quit();
+        }
+
         [TestMethod]
         public void SampleFunctionalTest1()
         {
-            try
-            {
-                driver = GetChromeDriver();
-                var webAppUrl = testContext.Properties["webAppUrl"].ToString();
-                driver.Navigate().GoToUrl(webAppUrl);
-                Assert.AreEqual("Home Page - My ASP.NET Application", driver.Title, "Expected title to be 'Home Page - My ASP.NET Application'");
-                var filePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()) + ".png";
-                var screenshot = driver.GetScreenshot();
-                screenshot.SaveAsFile(filePath);
-                testContext.AddResultFile(filePath);
-            }
-            finally
+            var numRetries = 5;
```

Comment:
Should we retry 12 times so as to wait for 1 min

Reply:
I think the webpage fetch itself is pretty time consuming. So 5 retries should be fine

Reply:
Added test timeout of 10 mins, with wait time reduced to 2 minutes